vertical-pod-autoscaler/pkg/recommender/routines/recommender_routines_test.go

```diff
@@ -0,0 +1,267 @@
+	config := model.GetAggregationsConfig()
+	maxAllowedMemory := 5e10
+	cpuSamples := []float64{1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0}
+	cpuHistogram := addTestCPUHistogram(cpuSamples, 1.0, testTimestamp, config.CPUHistogramOptions)
+
+	var memSamples []float64
+	for i := 0; i < 20; i++ {
+		memSamples = append(memSamples, 1e6)
+	}
+	memoryPeaksHistogram := addTestMemoryHistogram(memSamples, 1.0, testTimestamp, config.MemoryHistogramOptions)
+
+	ResourcePercentile := 0.9 // Same as in podResourceRecommender for target estimator
+	resourceEstimation := GetPercentileResourceEstimation(ResourcePercentile, ResourcePercentile,
+		cpuHistogram, memoryPeaksHistogram)
+	maxRelativeError := 0.05 // Allow 5% relative error to account for histogram rounding.
+	assert.InEpsilon(t, 1e7, model.BytesFromMemoryAmount(resourceEstimation[model.ResourceMemory]), maxRelativeError)
+
+	// Adding sudden surge sample to memory histogram
+	memoryPeaksHistogram.AddSample(4e10, 1.0, testTimestamp)
+	resourceEstimation = GetPercentileResourceEstimation(ResourcePercentile, ResourcePercentile,
+		cpuHistogram, memoryPeaksHistogram)
+
+	// still recommends 1e6 value as memory recommendation.
+	// Note: VPA recommender doesn't immediately respond to
+	// surge of memory resource
+	assert.InEpsilon(t, 1e7, model.BytesFromMemoryAmount(resourceEstimation[model.ResourceMemory]), maxRelativeError)
+
+	surgeSample := 4e10
+	// add a closure to increase memorySample by 1.2 times as defined in model.OOMBumpUpRatio
+	getOOMBumpUpRatio := func(memorySample float64) float64 { return memorySample * model.OOMBumpUpRatio }
+	resourceEstimation, surgeSamplesCount := surgeSampleBasedMemoryEstimation(surgeSample, maxAllowedMemory,
+		cpuHistogram, memoryPeaksHistogram, ResourcePercentile, ResourcePercentile, getOOMBumpUpRatio)
+
+	assert.InEpsilon(t, 60080658359, model.BytesFromMemoryAmount(resourceEstimation[model.ResourceMemory]), maxRelativeError)
+	assert.Greater(t, 3, surgeSamplesCount) // Fail if greater than 3 surge samples collected
+
+}
+
+// Verifies the number of samples it takes for the current
+// VPA recommender estimation to beat the current surge in usage.
+// The surge is determined and samples w.r.t that is added using
+// model.OOMMinBumpUp
+func TestAddMemorySurgeWithOOMMinBump(t *testing.T) {
+	config := model.GetAggregationsConfig()
+	maxAllowedMemory := 5e10
+	cpuSamples := []float64{1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0}
+	cpuHistogram := addTestCPUHistogram(cpuSamples, 1.0, testTimestamp, config.CPUHistogramOptions)
+
+	var memSamples []float64
+	for i := 0; i < 40; i++ {
+		memSamples = append(memSamples, 1e6)
+	}
+	memoryPeaksHistogram := addTestMemoryHistogram(memSamples, 1.0, testTimestamp, config.MemoryHistogramOptions)
+
+	ResourcePercentile := 0.9 // Same as in podResourceRecommender for target estimator
+	resourceEstimation := GetPercentileResourceEstimation(ResourcePercentile, ResourcePercentile,
+		cpuHistogram, memoryPeaksHistogram)
```

Comment:
Can't we call [podResourceRecommender.GetRecommendedPodResources](https://github.com/gardener/autoscaler/blob/a842944a648b7e3d87b2915f4c5fea077db9969b/vertical-pod-autoscaler/pkg/recommender/logic/recommender.go#L56:34) instead?

Reply:
ACK. Removed as not using estimator but using GetRecommendedPodResources now.